modules/bullet/shape_bullet.cpp

```diff
@@ -504,6 +504,8 @@ void HeightMapShapeBullet::set_data(const Variant &p_data) {
 	int l_width = d["width"];
 	int l_depth = d["depth"];
 
+	ERR_FAIL_COND(l_width < 2 || l_depth < 2);
```

Comment:
Since it can be caused by invalid user input, there should be a `_MSG` explanation I think.

Reply:
I agree, but there are currently no messages for the rest of the checks in this function. I can add messages to them too as part of this PR, but I thought it best to focus on fixing the issue and leave adding messages to a separate PR.

Reply:
I haven't looked into details, but given the API of `HeightMapShape3D`:
```
                <member name="map_data" type="PackedFloat32Array" setter="set_map_data" getter="get_map_data" default="PackedFloat32Array( 0, 0, 0, 0 )">
                        Height map data, pool array must be of [member map_width] * [member map_depth] size.
                </member>
                <member name="map_depth" type="int" setter="set_map_depth" getter="get_map_depth" default="2">
                        Depth of the height map data. Changing this will resize the [member map_data].
                </member>
                <member name="map_width" type="int" setter="set_map_width" getter="get_map_width" default="2">
                        Width of the height map data. Changing this will resize the [member map_data].
                </member>
```

I think that the validation of the Dictionary's content is validating internal engine code, so the previous checks are more sanity checks for engine developers (if they fail, it means there's a bug in the engine, and thus we don't need a clear message as the source code is the clearest we have already).

But this one here can fail based on user input (e.g. `map_width = 1`), so this needs to be handled with a user-facing explanation. At least that's where I set the cursor myself.

Reply:
Though it's true that `ERR_FAIL_COND(l_min_height > l_max_height);` might also be something that can be triggered by user scripts (again, didn't look into implementation details to confirm).

Reply:
Updated with `ERR_FAIL_COND_MSG`.